sdk/storage/azblob/internal/exported/blob_batch.go

```diff
@@ -0,0 +1,269 @@
+//	x-ms-date: Thu, 14 Jun 2018 16:46:54 GMT
+//	Authorization: SharedKey account:G4jjBXA7LI/RnWKIOQ8i9xH4p76pAQ+4Fs4R1VxasaE=
+//	Content-Length: 0
+func CreateBatchRequest(bb *BlobBatchBuilder) ([]byte, string, error) {
+	batchID, err := createBatchID()
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Create a new multipart buffer
+	reqBody := &bytes.Buffer{}
+	writer := multipart.NewWriter(reqBody)
+
+	// Set the boundary
+	err = writer.SetBoundary(batchID)
+	if err != nil {
+		return nil, "", err
+	}
+
+	partHeaders := make(textproto.MIMEHeader)
+	partHeaders["Content-Type"] = []string{"application/http"}
+	partHeaders["Content-Transfer-Encoding"] = []string{"binary"}
+	var partWriter io.Writer
+
+	for i, req := range bb.SubRequests {
+		if bb.AuthPolicy != nil {
+			resp, err := bb.AuthPolicy.Do(req)
+			if err != nil && resp != nil {
+				if log.Should(EventSubmitBatch) {
+					log.Writef(EventSubmitBatch, "failed to authorize sub-request for %v.\nError: %v\nResponse status: %v", req.Raw().URL.Path, err.Error(), resp.Status)
+				}
+			}
+		}
+
+		partHeaders["Content-ID"] = []string{fmt.Sprintf("%v", i)}
+		partWriter, err = writer.CreatePart(partHeaders)
+		if err != nil {
+			return nil, "", err
+		}
+
+		_, err = partWriter.Write(buildSubRequest(req))
+		if err != nil {
+			return nil, "", err
+		}
+	}
+
+	// Close the multipart writer
+	err = writer.Close()
+	if err != nil {
+		return nil, "", err
+	}
+
+	return reqBody.Bytes(), batchID, nil
+}
+
+// UpdateSubRequestHeaders updates the sub-request headers.
+// Removes x-ms-version header.
+func UpdateSubRequestHeaders(req *policy.Request) {
+	// remove x-ms-version header from the request header
+	for k := range req.Raw().Header {
+		if strings.EqualFold(k, shared.HeaderXmsVersion) {
+			delete(req.Raw().Header, k)
+		}
+	}
+}
+
+// BlobBatchSubResponse contains the response for the individual sub-requests.
+type BlobBatchSubResponse struct {
+	ContentID     *int
+	ContainerName *string
+	BlobName      *string
+	RequestID     *string
+	Version       *string
+	Error         error // nil error indicates that the batch sub-request operation is successful
+}
+
+func getResponseBoundary(contentType *string) (string, error) {
+	if contentType == nil {
+		return "", fmt.Errorf("Content-Type returned in SubmitBatch response is nil")
+	}
+	boundaryIdx := strings.Index(*contentType, "batchresponse_")
+	if boundaryIdx == -1 {
+		return "", fmt.Errorf("batch boundary not present in Content-Type header of the SubmitBatch response.\nContent-Type: %v\n", *contentType)
+	}
+	return (*contentType)[boundaryIdx:], nil
+}
+
+func getContentID(part *multipart.Part) (*int, error) {
+	contentID := getResponseHeader("Content-ID", part.Header)
+	if contentID == nil {
+		return nil, nil
+	}
+
+	val, err := strconv.Atoi(strings.TrimSpace(*contentID))
+	if err != nil {
+		return nil, err
+	}
+	return &val, nil
+}
+
+func getResponseHeader(key string, headers map[string][]string) *string {
```

Comment:
I don't think this is necessary.  Can't you use the `Get()` method on `http.Header`?

Reply:
`Get(key)` method on `http.Header` canonicalizes the key. This was why I have the method `getResponseHeader()` in place of that. I will check if `Get(key)` can used instead of this method.

Reply:
Updated the logic to use `Get()` method of `http.Header` here.